### PR TITLE
use basaran for serving for falcon-7b-instruct

### PR DESCRIPTION
**Pull request by samos123**
Basaran provides both streaming UI to show tokens as they are being generated and also provides an OpenAI compatible endpoint.

### CODE REVIEW

**Comment by nstogner** [comment]
Looks like basaran uses SSE (server sent events): https://github.com/hyperonym/basaran/blob/0c4b8a597548e3f6242129999bd4114085541c06/basaran/__main__.py#L212

File: falcon-7b-instruct/requirements.txt [hunk 1]
```diff
@@ -5,6 +5,7 @@ torch
 xformers
 notebook
 datasets
+basaran
```
Comment (nstogner) [inline falcon-7b-instruct/requirements.txt:8]: Basaran also comes with additional endpoints that dont make a ton of sense with our arch... https://github.com/hyperonym/basaran/blob/0c4b8a597548e3f6242129999bd4114085541c06/basaran/__main__.py#L113

Perhaps we dont worry about this... but just wanted to point it out.

File: falcon-7b-instruct/Dockerfile [hunk 2]
```diff
@@ -24,5 +29,5 @@ RUN --mount=type=cache,target=/root/.cache pip install --no-cache-dir -r require
 COPY *.py .
 COPY sample-data sample-data
 
-CMD python serve.py
+CMD PORT=8080 python -m basaran
```
[the remark this answers is not in the document]
Comment (nstogner) [reply falcon-7b-instruct/Dockerfile:32]: I lean towards the model contract being:

`serve.sh` & `train.sh`

[the remark this answers is not in the document]
Comment (nstogner) [reply falcon-7b-instruct/Dockerfile:32]: Perhaps those are the defaults, and the Model object allows for over-riding them.

[the remark this answers is not in the document]
Comment (samos123) [reply falcon-7b-instruct/Dockerfile:32]: +1 to having serve.sh and train.sh. I'm not so sure for the use case of infer.sh? Does that need to be a contract?

File: falcon-7b-instruct/serve.sh [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+
```
Comment (nstogner) [inline falcon-7b-instruct/serve.sh:2]: I think `set -e` is needed.

**Review by nstogner — CHANGES_REQUESTED** [review]

File: falcon-7b-instruct/serve.sh (hunk 3 above)
Comment (samos123) [reply falcon-7b-instruct/serve.sh:2]: fail if something goes wrong is a good idea. Will add.

Comment (samos123) [reply falcon-7b-instruct/serve.sh:2]: are you ok with setting `set -xe`?

Comment (samos123) [reply falcon-7b-instruct/serve.sh:2]: done

**Comment by samos123** [comment]
merging since change requested was done and want to test it out in substratus end to end

**Pull request merged**